compiler/rustc_typeck/src/check/method/suggest.rs

```diff
@@ -1270,30 +1270,45 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         None
     }
 
+    fn get_trait_ref_self_type(&self, trait_pred: ty::TraitPredicate<'tcx>) -> Ty<'tcx> {
+        fn get_self_type<'a>(tcx: TyCtxt<'_>, ty: Ty<'a>) -> Ty<'a> {
+            match ty.kind() {
+                ty::Adt(def, subs) => {
+                    if tcx.is_diagnostic_item(sym::Vec, def.did()) {
+                        return get_self_type(tcx, subs.type_at(0));
+                    }
+                    ty
+                }
+                ty::Array(ty, _) => {
+                    return get_self_type(tcx, *ty);
+                }
+                _ => ty,
+            }
+        }
+
+        if Some(trait_pred.trait_ref.def_id) == self.tcx.lang_items().eq_trait() {
```

Comment:
Hmm. In case of `PartialOrd`, rustc properly explains why `xs` can't be compared with [X, X, X]. 

```
fn main() {
    struct X;
    let xs = [X, X, X];
    let eq = xs > [X, X, X];
}
```

https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=7115dbcd188f44a242ba8640347bbfe3

Reply:
Is it possible to implement the `PartialEq` suggestion by re-using the code for `PartialOrd`?